src/network.py

```diff
@@ -194,7 +238,7 @@ def decode_sequence(self, target, hidden_state, enc_output, enc_ctxt_mask, teach
             if random.random() < teacher_ratio and i + 1 < max_out_len:
                 input_ = target[:, i+1, :].unsqueeze(1)
             else:
-                input_ = outputs[-1]
+                input_ = outputs[-1].detach().permute(0, 2, 1)
```

Comment:
Same here. Why are the gradients detached here?

Also, if I'm understanding this right, are we using only the last output, at time t, of the decoder as input to the decoder for the next step, t + 1, instead of a sequence of outputs from 1 to t?

Reply:
The comment on detached can be ignored, but the decoding question remains here.

Reply:
In the case we are using autoregressive input, we have to detach() the gradient here - if it does not get detached, we get NaN's (I think this is because input is an output of the LSTM, so feeding it back in leads to some error/overlapping gradients?)